src/main/java/hudson/plugins/jira/selector/DefaultIssueSelector.java

```diff
@@ -128,7 +128,7 @@ protected void addIssuesFromCurrentBuild(Run<?, ?> build, JiraSite site, TaskLis
      */
     protected void addIssuesFromDependentBuilds(Run<?, ?> build, JiraSite site, TaskListener listener,
             Set<String> issueIds) {
-        for (DependencyChange depc : RunScmChangeExtractor.getDependencyChanges(build.getPreviousBuild()).values()) {
+        for (DependencyChange depc : RunScmChangeExtractor.getDependencyChanges(build).values()) {
```

Comment:
Previously it was getting dependency changes between previous build and ... previous build. 
Now it is taking dependency changes between current and previous build.